v8.15.0 breaks things related to type hints?

Hello,

some of my existing code breaks when upgrading from v8.14.0 to v8.15.0 as follows:

```
self = elasticsearch_dsl.utils.AttrDict[typing.Any], attr = 'doc_count'

    def __getattr__(self, attr):
        if attr in {'__name__', '__qualname__'}:
            return self._name or self.__origin__.__name__

        # We are careful for copy and pickle.
        # Also for simplicity we don't relay any dunder names
        if '__origin__' in self.__dict__ and not _is_dunder(attr):
>           return getattr(self.__origin__, attr)
E           AttributeError: type object 'AttrDict' has no attribute 'doc_count'

/usr/lib/python3.12/typing.py:1185: AttributeError
```

Greetings,

jgb

--- a/elasticsearch_dsl/utils.py
+++ b/elasticsearch_dsl/utils.py
@@ -215,7 +215,11 @@ class AttrDict(Generic[_ValT]):
         del self._d_[key]
 
     def __setattr__(self, name: str, value: _ValT) -> None:
-        if name in self._d_ or not hasattr(self.__class__, name):
+        # the __orig__class__ attribute has to be treated as an exception, as
+        # is it added to an object when it is instantiated with type arguments
+        if (
+            name in self._d_ or not hasattr(self.__class__, name)
+        ) and name != "__orig_class__":
             self._d_[name] = value
         else:
             # there is an attribute on the class (could be property, ..) - don't add it as field

--- a/tests/test_utils.py
+++ b/tests/test_utils.py
@@ -45,10 +45,21 @@ def test_attrlist_slice() -> None:
     assert isinstance(l[:][0], MyAttrDict)
 
 
+def test_attrlist_with_type_argument() -> None:
+    a = utils.AttrList[str](["a", "b"])
+    assert list(a) == ["a", "b"]
+
+
 def test_attrdict_keys_items() -> None:
     a = utils.AttrDict({"a": {"b": 42, "c": 47}, "d": "e"})
     assert list(a.keys()) == ["a", "d"]
     assert list(a.items()) == [("a", {"b": 42, "c": 47}), ("d", "e")]
+
+
+def test_attrdict_with_type_argument() -> None:
+    a = utils.AttrDict[str]({"a": "b"})
+    assert list(a.keys()) == ["a"]
+    assert list(a.items()) == [("a", "b")]
 
 
 def test_merge() -> None:

